Userland/Libraries/LibWeb/HTML/HTMLElement.cpp

```diff
@@ -242,8 +242,8 @@ static void run_focus_update_steps(NonnullRefPtrVector<DOM::Node> old_chain, Non
     (void)new_focus_target;
 
     // 4. For each entry entry in new chain, in reverse order, run these substeps:
-    for (ssize_t i = new_chain.size() - 1; i >= 0; --i) {
-        auto& entry = new_chain[i];
+    for (auto it = new_chain.rbegin(); it != new_chain.rend(); ++it) {
+        auto& entry = *it;
```

Comment:
This feels like a very mild upgrade/sidegrade in ergonomics.
How about we add some kind of reversal adapter thing so that something this becomes possible:
```cpp
for (auto& entry : new_chain.in_reverse()) {
    ...
```

Reply:
This is just for "first" usage, I saw your video you were searching about reverse iterator :) also can be used in many other places.
Btw I can add `in_reverse()` function

Reply:
We already have 
```
    void reverse()
    {
        for (size_t i = 0; i < size() / 2; ++i)
            AK::swap(at(i), at(size() - i - 1));
    }
```

But reverse iterator doesn't edit the vector, it is only more clear that you are running in reverse :)

Reply:
I don't mean that we should add an API that reverses the container. I mean we should add an adapter template that allows us to use range-for to iterate a container in reverse :)

Reply:
Ok thanks for advice :)
I implement it and seems nicer :^)